Source/JNA/waffle-jna/src/main/java/waffle/servlet/WindowsPrincipal.java

```diff
@@ -255,4 +255,30 @@ public String toString() {
         return this.getName();
     }
 
+    /*
+     * (non-Javadoc)
+     * @see java.lang.Object#equals(java.lang.Object)
+     */
+    @Override
+    public boolean equals(final Object o) {
```

Comment:
It appears it could be but that just stops extending it and overriding methods in it.  Since it doesn't contain anything to change mutability other than a new instance it's probably ok.  I'll have to look further to see why it has two non final fields.  Feel free to investigate a bit further.  Thanks.


Sent from Outlook Mobile






On Mon, Jan 11, 2016 at 12:13 PM -0800, "Sergey Podolsky" <notifications@github.com> wrote:





> @@ -255,4 +255,30 @@ public String toString() {
>          return this.getName();
>      }
>
> +    /*
> +     * (non-Javadoc)
> +     * @see java.lang.Object#equals(java.lang.Object)
> +     */
> +    @Override
> +    public boolean equals(final Object o) {

Shouldn't the class be marked as final for immutability?

---
Reply to this email directly or view it on GitHub:
https://github.com/dblock/waffle/pull/310/files#r49372072